### PR TITLE
Sets initial crontab for schedule

### LINKED ISSUE #180: If a user selects "Custom interval" right away, there is no default value.
## Description

If a user selects "Custom interval" without making other changes to the schedule, there is no default value in the cron expression text field.

## Reproduce

1. Create a job.
2. Add a schedule, but immediately change to "Custom interval"
3. cron expression will be empty.

## Expected behavior

Cron expression should always have a default value.

## Context

- Version: 0.5.1

### PR DESCRIPTION
**Pull request by jweill-aws**
Fixes #180. When the user sets up an initial schedule with "Weekday" mode, there is now an initial value for the cron expression, which is used when the user switches to custom scheduling.

https://user-images.githubusercontent.com/93281816/197264307-88b565e8-c364-45a6-8fca-2704697c4bc9.mov

### CODE REVIEW

File: src/model.ts [hunk 1]
```diff
@@ -279,8 +284,8 @@ export function convertDescribeJobtoJobDetail(
     updateTime: describeJob.update_time,
     startTime: describeJob.start_time,
     endTime: describeJob.end_time,
-    scheduleInterval: 'weekday',
-    downloaded: describeJob.downloaded
+    downloaded: describeJob.downloaded,
+    ...InitialScheduleOptions
```
Comment (dlqqq) [inline src/model.ts:288]: The describe job model should not have any concept of a schedule. It looks like the root cause of this is the fact that `IJobDetailModel` is extending `ICreateJobModel`, which I assume is due to just not wanting to rewrite the types for each property again.

While inheriting the `ICreateJobModel` is shorter, it is fundamentally incorrect as shown here. What's confusing is that although the `DescribeJob` response inherits from the `CreateJob` response (which we are confusingly also calling models), the `IDescribeJobModel` contains the UI state of the `DescribeJob` page, which does not necessarily share all the UI state of the `CreateJob` page. This is part of the model refactor I am proposing here: https://github.com/jupyter-server/jupyter-scheduler/issues/105

To fix this, you should change `IDescribeJobModel` to not inherit from `ICreateJobModel`, and then remove any `ICreateJobModel`-specific properties from this object you are returning here.

File: src/model.ts [hunk 2]
```diff
@@ -59,6 +59,12 @@ export interface ICreateJobModel extends PartialJSONObject {
   scheduleWeekDay?: string;
 }
 
+export const InitialScheduleOptions = {
+  scheduleInterval: 'weekday',
+  schedule: '0 0 * * MON-FRI',
+  timezone: Intl.DateTimeFormat().resolvedOptions().timeZone
+};
+
```
Comment (dlqqq) [inline src/model.ts:67]: You don't need a separate object here. You should add `schedule: '0 0 * * MON-FRI'` into `emptyCreateJobModel()`. Then, whenever building a new create job model in the front end (i.e. in refill button or rerun job), you should pass the properties of this "empty" (which is actually confusingly non-empty) model via destructuring like so:

```
    const initialState: ICreateJobModel = {
      ...emptyCreateJobModel(),
      jobName: props.model.jobName,
      inputFile: props.model.inputFile,
      outputPath: props.model.outputPrefix ?? '',
      environment: props.model.environment,
      runtimeEnvironmentParameters: props.model.runtimeEnvironmentParameters,
      parameters: props.model.parameters,
      outputFormats: props.model.outputFormats,
    }
```

Renaming these model generators from `empty*` to `default*` or some other more semantically accurate name will be part of https://github.com/jupyter-server/jupyter-scheduler/issues/105. No need to do so here.

**Review by dlqqq — CHANGES_REQUESTED** [review]

File: src/model.ts (hunk 1 above)
Comment (jweill-aws) [reply src/model.ts]: This is now done, with additional modifications to the advanced options to change expectations for the nature of the `model` prop.

**Comment by jweill-aws** [comment]
FYI @awaisabir: If this is merged, your advanced options extension might need to be modified for create and detail views.

**Comment by awaisabir** [comment]
> FYI @awaisabir: If this is merged, your advanced options extension might need to be modified for create and detail views.

What modification are we talking about exactly? Are you saying we need to allow users to pass in idempotency tokens as well? Sorry a bit confused here

**Comment by dlqqq** [comment]
@awaisabir Basically the type of `model` being passed to AdvancedOptions is slightly changing because we had some extra properties that shouldn't be there. I wouldn't worry about this since you shouldn't be using any of those attributes. But we just wanted to give you a heads up so you know what might be wrong if you find out the types for your advanced options extension break.

**Review by dlqqq — APPROVED** [review]

**Comment by awaisabir** [comment]
> @awaisabir Basically the type of `model` being passed to AdvancedOptions is slightly changing because we had some extra properties that shouldn't be there. I wouldn't worry about this since you shouldn't be using any of those attributes. But we just wanted to give you a heads up so you know what might be wrong if you find out the types for your advanced options extension break.

Alright. Thanks!

**Pull request merged**